### PR TITLE
Fixed colormin function and added colormin tests

### PR DESCRIPTION
**Pull request by TrySound**

### CODE REVIEW

**Comment by ben-eb** [comment]
package.json should have the `main` key as `dist/index.js` and the `files` array should have the `dist` folder. `dist` should also go into `.gitignore`. :smile:

File: src/__tests__/index.js [hunk 1]
```diff
@@ -32,11 +32,11 @@ var tests = module.exports = [{
 }, {
     message: 'should minify color values in background gradients',
     fixture: 'h1{background:linear-gradient( #ff0000,yellow )}',
-    expected: 'h1{background:linear-gradient(red,#ff0)}'
+    expected: 'h1{background:linear-gradient( red,#ff0 )}'
 }, {
     message: 'should minify color values in background gradients (2)',
     fixture: 'h1{background:linear-gradient(yellow, orange), linear-gradient(black, rgba(255, 255, 255, 0))}',
-    expected: 'h1{background:linear-gradient(#ff0, orange), linear-gradient(#000, hsla(0,0%,100%,0))}'
+    expected: 'h1{background:linear-gradient(#ff0, orange), linear-gradient(#000, transparent)}'
```
Comment (ben-eb) [inline src/__tests__/index.js]: This is incorrect; the `transparent` keyword means 0% black, whereas that test is 0% white. It makes a difference in terms of the color stops in gradients, so this should output the hsla value.

**Comment by ben-eb** [comment]
Don't forget that we have to make an exception for `-webkit-tap-highlight-color`. https://github.com/ben-eb/postcss-colormin/issues/1

```diff
-.ui.form input:not([type]),.ui.form input[type=date],.ui.form input[type=datetime-local],.ui.form input[type=email],.ui.form input[type=number],.ui.form input[type=password],.ui.form input[type=search],.ui.form input[type=tel],.ui.form input[type=text],.ui.form input[type=time],.ui.form input[type=url],.ui.form textarea{
-  margin:0;outline:none;-webkit-appearance:none;tap-highlight-color:hsla(0,0%,100%,0);font-size:1em;background:#fff;border:1px solid rgba(34,36,38,.15);color:rgba(0,0,0,.87);border-radius:.28571429rem;box-shadow:0 0 0 0 transparent inset;-webkit-transition:color .1s ease,border-color .1s ease;transition:color .1s ease,border-color .1s ease
-}
+.ui.form input:not([type]),.ui.form input[type=date],.ui.form input[type=datetime-local],.ui.form input[type=email],.ui.form input[type=number],.ui.form input[type=password],.ui.form input[type=search],.ui.form input[type=tel],.ui.form input[type=text],.ui.form input[type=time],.ui.form input[type=url],.ui.form textarea{
+  margin:0;outline:none;-webkit-appearance:none;tap-highlight-color:transparent;font-size:1em;background:#fff;border:1px solid rgba(34,36,38,.15);color:rgba(0,0,0,.87);border-radius:.28571429rem;box-shadow:0 0 0 0 transparent inset;-webkit-transition:color .1s ease,border-color .1s ease;transition:color .1s ease,border-color .1s ease
+}
```

File: src/lib/colormin.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            hsla[1] = hsla[1] + '%';
+            hsla[2] = hsla[2] + '%';
+            return shorter(`hsla(${hsla.join()})`, `rgba(${rgba.join()})`);
+        }
+    }
+
+    if (isHex(word)) {
+        word = toLonghand(word);
+        if (word in hexes) {
+            return shorter(toShorthand(word), hexes[word]);
```
Comment (ben-eb) [inline src/lib/colormin.js]:
```js
return shorter(hexes[word], toShorthand(word));
```

preserves the original behaviour.

File: src/lib/colormin.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        }
+        return toShorthand(word);
+    }
+
+    if (word in keywords) {
+        return shorter(word, toShorthand(keywords[word]));
```
Comment (ben-eb) [inline src/lib/colormin.js]:
```js
return shorter(toShorthand(keywords[word]), word);
```

preserves the original behaviour.

**Comment by ben-eb** [comment]
Issues are resolved, save for a module test in cssnano.

```
    ✖ should process multiple backgrounds
    --------------------------------------
      operator: equal
      expected: |-
        'h1{background:url(foo/bar.jpg),url(http://website.com/img.jpg)}'
      actual: |-
        'h1{background:url(foo/bar.jpg), url(http://website.com/img.jpg)}'
```

Seems like this module did more than initially realised.

**Comment by TrySound** [comment]
@ben-eb It's cssnano/postcss-normalize-url fix, isn't it?

**Pull request merged**